### PR TITLE
ACDD refactor date

### PR DESCRIPTION
**Pull request by benjwadams**
Alters priority of tests relating to ISO date global attributes in ACDD 1.3.

Refactors handling of global attribute lists to use an `OrderedDict` for the underlying operations.

Addresses #519

### CODE REVIEW

File: compliance_checker/base.py [hunk 1]
```diff
@@ -246,81 +247,19 @@ def attr_check(l, ds, priority, ret_val, gname=None):
...
-                ret_val.append(
-                    Result(
-                        priority,
-                        res,
-                        gname if gname else name,
-                        msgs
-                    )
-                )
-            else:
-                ret_val.append(other(ds)(priority))
```
Comment (benjwadams) [inline compliance_checker/base.py]: Oops, this could be just `ret_val.append(res(priority))`

File: compliance_checker/util.py [hunk 2]
```diff
@@ -33,3 +34,9 @@ def dateparse(date_str):
 
     return pendulum.parse(date_str)
 
+def kvp_convert(input_coll):
```
Comment (benjwadams) [inline compliance_checker/util.py]: I should add a docstring describing the purpose of this helper method.

**Comment by benjwadams** [comment]
Above comments should be addressed.  Please review updated code.

File: compliance_checker/acdd.py [hunk 3]
```diff
@@ -634,6 +636,28 @@ def __init__(self):
...
+                                      result_name, msgs)
+
+        # run ISO 8601 date checks against the date_created, date_issued,
+        # date_modified, and date_metadata_modified global attributes
+        self.rec_atts = kvp_convert(self.rec_atts)
+        self.rec_atts['date_created'] = partial(_check_attr_is_iso_date,
+                                                'date_created')
+        self.sug_atts = kvp_convert(self.sug_atts)
+        for k in ("date_{}".format(suffix) for suffix in
+                  ('issued', 'modified', 'metadata_modified')):
```
Comment (Bobfrat) [inline compliance_checker/acdd.py]: Maybe i'm old-school but this seems unnecessarily complex.

In my opinion, something like this is easier to follow:
```
date_attrs = ['date_issued', 'date_modified', 'date_metadata_modified']
for date_attr in date_attrs:
```

**Pull request merged**